core/foundation/inc/RtypesCore.h

```diff
@@ -34,91 +45,91 @@ class TRootIOCtor;
+constexpr Int_t kMaxChar = kMaxUChar >> 1;///< \deprecated Consider replacing with `std::numeric_limits<char>::max()` (or `std::int8_t`)
+constexpr Int_t kMinChar = -kMaxChar - 1; ///< \deprecated Consider replacing with `std::numeric_limits<char>::lowest()` (or `std::int8_t`)
 
-constexpr Int_t kMaxUShort = UShort_t(~0);
-constexpr Int_t kMaxShort = kMaxUShort >> 1;
-constexpr Int_t kMinShort = -kMaxShort - 1;
+constexpr Int_t kMaxUShort = UShort_t(~0);  ///< \deprecated Consider replacing with `std::numeric_limits<unsigned short>::max()` (or `std::uint16_t`)
+constexpr Int_t kMaxShort = kMaxUShort >> 1;///< \deprecated Consider replacing with `std::numeric_limits<short>::max()` (or `std::int16_t`)
+constexpr Int_t kMinShort = -kMaxShort - 1; ///< \deprecated Consider replacing with `std::numeric_limits<short>::lowest()` (or `std::int16_t`)
 
-constexpr UInt_t kMaxUInt = UInt_t(~0);
-constexpr Int_t kMaxInt = Int_t(kMaxUInt >> 1);
-constexpr Int_t kMinInt = -kMaxInt - 1;
+constexpr UInt_t kMaxUInt = UInt_t(~0);        ///< \deprecated Consider replacing with `std::numeric_limits<unsigned int>::max()` (or `std::uint32_t`)
+constexpr Int_t kMaxInt = Int_t(kMaxUInt >> 1);///< \deprecated Consider replacing with `std::numeric_limits<int>::max()` (or `std::int32_t`)
+constexpr Int_t kMinInt = -kMaxInt - 1;        ///< \deprecated Consider replacing with `std::numeric_limits<int>::lowest()` (or `std::int32_t`)
 
-constexpr ULong_t kMaxULong = ULong_t(~0);
-constexpr Long_t kMaxLong = Long_t(kMaxULong >> 1);
-constexpr Long_t kMinLong = -kMaxLong - 1;
+constexpr ULong_t kMaxULong = ULong_t(~0);         ///< \deprecated Consider replacing with `std::numeric_limits<unsigned long>::max()`
+constexpr Long_t kMaxLong = Long_t(kMaxULong >> 1);///< \deprecated Consider replacing with `std::numeric_limits<long>::max()`
+constexpr Long_t kMinLong = -kMaxLong - 1;         ///< \deprecated Consider replacing with `std::numeric_limits<long>::lowest()`
 
-constexpr ULong64_t kMaxULong64 = ULong64_t(~0LL);
-constexpr Long64_t kMaxLong64 = Long64_t(kMaxULong64 >> 1);
-constexpr Long64_t kMinLong64 = -kMaxLong64 - 1;
+constexpr ULong64_t kMaxULong64 = ULong64_t(~0LL);         ///< \deprecated Consider replacing with `std::numeric_limits<unsigned long long>::max()` (or `std::uint64_t`)
+constexpr Long64_t kMaxLong64 = Long64_t(kMaxULong64 >> 1);///< \deprecated Consider replacing with `std::numeric_limits<long long>::max()` (or `std::int64_t`)
+constexpr Long64_t kMinLong64 = -kMaxLong64 - 1;           ///< \deprecated Consider replacing with `std::numeric_limits<long long>::lowest()` (or `std::int64_t`)
 
-constexpr ULong_t kBitsPerByte = 8;
-constexpr Ssiz_t kNPOS = ~(Ssiz_t)0;
+constexpr ULong_t kBitsPerByte = 8; ///< \deprecated Consider replacing with `CHAR_BIT` from `<climits>`
+constexpr Ssiz_t kNPOS = ~(Ssiz_t)0;///< \deprecated Consider replacing with `int(-1)` or `std::string::npos`
```

Comment:
Why deprecated this one?  It is the `TString` marker and it is confused to ask the user to hard code or to use a different marker (that may or may not stay the same number).